File: Stack/stack.py
#An implementation of the stack datastructure in Python
class Stack:
    def __init__(self, max_size):
        self.stack = []
        self.max_size = max_size

    def push_stack(self, val):
        if not self.isFull():
            self.stack.append(val)
        else:
            print("Stack Full")

    def pop_stack(self):
        return self.stack.pop()

    def top(self):
        return self.stack[-1]

    @property
    def isEmpty(self):
        return len(self.stack) == 0

    def isFull(self):
        return len(self.stack) == self.max_size

File: Stack/stack.js
// Stack implementation in JavaScript
class Stack {
    constructor(max_size) {
        this.stack = [];
        this.max_size = max_size;
    };

    push_stack(val) {
        if (!this.isFull()) {
            this.stack.push(val);
        }
        else {
            console.log("Stack Full")
        }
    };

    pop_stack() {
        return this.stack.pop();
    };

    top_stack() {
        return this.stack[this.stack.length - 1];
    };

    isFull() {
        return this.stack.length == this.max_size;
    };

    isEmpty() {
        return this.stack.length == 0;
    };
}
